src/actions/roleMeWithMenu.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  MessageActionRow,
+  MessageSelectMenu,
+  MessageSelectOptionData,
+} from "discord.js";
+
+const { SlashCommandBuilder } = require("@discordjs/builders");
+
+//Work in Progress , 
+// ** should only show with refactored deploy-commands.ts
+/**
+ * Main issue is that this list is too large for people to look at quickly
+ * Something to break these classes into groups would help
+ * 
+ * Overloading the command is also an option
+ * Example ->
+ * /role me [Name] [Class]
+ * /role me [Name] [Class] [Class]
+ * /role me [Name] [Class] [Class] [Class]
+ * /role me [Name] [Class] [Class] [Class] [Class]
+ */
+module.exports = {
+  data: new SlashCommandBuilder().setName("roleMe").setDescription("WIP")
+  .addStringOption((option :any) =>
+  option
+    .setName("name")
+    .setDescription("Real Name")
+    .setRequired(true)
+),
+  async execute(interaction: any, client: any) {
+    const row = new MessageActionRow().addComponents(
+      new MessageSelectMenu()
+        .setCustomId("classSelect")
+        .setPlaceholder("Please Select a Class")
+        .setMinValues(1)
+        .setMaxValues(5) //This can be anything really, but I wouldn't expect a student to have more than 5
+        .addOptions(
+            //Instead of creating each option by hand, pull from a list and create the objects at runtime
+            //!Not great for efficiency
+          classes.map((c: string) => {
+            return {
+              label: c,
+              value: c,
+            };
+          })
+        )
+    );
+    await interaction.reply({
+      content: "Choose Your Classes. ",
+      components: row,
+    });
+  },
+};
+
+// This should be a constant somewhere
+const classes: string[] = [
+  "CFS-262",
```

Comment:
I disagree because the list is mutable. It will likely be added to on a semester basis